pybotx_widgets/checktable.py

```diff
@@ -42,71 +41,86 @@ def is_checkbox_value_exist_in_mapping(cls, values):
-            checkbox_status = strings.CHECKBOX_UNCHECKED
-        else:
-            checkbox_status = strings.CHECKBOX_CHECKED
+        if checkbox.mapping:
+            if is_undefined:
+                return self.CHOOSE_LABEL
+            return checkbox.mapping[checkbox.checkbox_value]
 
-        checkbox_text = f"{checkbox_status} {checkbox.label}"
-
-        value_text = get_value_text(checkbox)
+        if is_undefined:
+            return self.FILL_LABEL
 
-        markup.add_bubble(uncheck_command, checkbox_text, data=checkbox.data)
-        markup.add_bubble(
-            checkbox.command, value_text, new_row=False, data=checkbox.data
-        )
+        return checkbox.checkbox_value
 
-    return markup
+    def add_checkboxes(self) -> None:
+        """Add checkbox."""
 
+        for checkbox in self.checkboxes:
+            checkbox.data = checkbox.data or {}
 
-async def checktable(
-    message: Message,
-    bot: Bot,
-    checkboxes: List[CheckboxContent],
-    label: str,
-    uncheck_command: str,
-    additional_markup: MessageMarkup = None,
-) -> None:
-    """Create checktable widget.
+            if isinstance(checkbox.checkbox_value, Undefined):
+                checkbox_status = self.CHECKBOX_UNCHECKED
+            else:
+                checkbox_status = self.CHECKBOX_CHECKED
 
-    :param message - botx Message
-    :param bot - botx Bot
-    :param checkboxes - All content to be displayed
-    :param label - Text of message
-    :param uncheck_command - Command for handler which uncheck value
-    :param additional_markup - Additional markup for attaching to widget
-    """
+            checkbox_text = f"{checkbox_status} {checkbox.label}"
 
-    markup = add_checkboxes(checkboxes, uncheck_command)
+            value_text = self.get_button_value_text(checkbox)
 
-    if additional_markup:
-        markup = merge_markup(markup, additional_markup)
+            self.markup.add_bubble(self.uncheck_command, checkbox_text, data=checkbox.data)
+            self.markup.add_bubble(
+                checkbox.command, value_text, new_row=False, data=checkbox.data
+            )
 
-    await send_or_update_message(message, bot, label, markup)
+    def add_markup(self) -> None:
+        self.add_checkboxes()
+        self.add_additional_markup()
+
+
+class ChecktableWidget(Widget, MarkupMixin):
+    def __init__(
+        self,
+        checkboxes: List[CheckboxContent],
+        label: str,
+        uncheck_command: str,
+        *args: Any,
+        **kwargs: Any
+    ):
+        """Create checktable widget.
+
+        :param checkboxes - All content to be displayed
+        :param label - Text of message
+        :param uncheck_command - Command for handler which uncheck value
+        """
+        super().__init__(*args, **kwargs)
+
+        self.checkboxes = checkboxes
+        self.label = label
+        self.uncheck_command = uncheck_command
+
+    async def send_widget_message(self):
```

Comment:
Mb make this function private ( `_send_widget_message` or even `__send_widget_message`)?  
It is unlikely that anyone will call this function directly

Reply:
Widgets often need to be customized, because of that, i made all methods open, so that anyone can override them without linter errors.